plugins/hls-rename-plugin/src/Ide/Plugin/Rename.hs

```diff
@@ -65,16 +64,25 @@ descriptor pluginId = (defaultPluginDescriptor pluginId)
 renameProvider :: PluginMethodHandler IdeState TextDocumentRename
 renameProvider state pluginId (RenameParams (TextDocumentIdentifier uri) pos _prog newNameText) =
     pluginResponse $ do
-        nfp <- safeUriToNfp uri
-        oldName <- getNameAtPos state nfp pos
-        refLocs <- refsAtName state nfp oldName
+        nfp <- handleUriToNfp uri
+        directOldNames <- getNamesAtPos state nfp pos
+        directRefs <- concat <$> mapM (refsAtName state nfp) directOldNames
+
+        -- References of references to account for punned names (Indirect references)
```

Comment:
You wrote a lovely explanation of this in the PR description that's going to probably get lost when you merge the PR, maybe move it into a longer comment in the code?